xcodeproj/internal/top_level_targets.bzl

```diff
@@ -425,6 +425,18 @@ def process_top_level_target(
             )
             for mergeable_target in mergeable_targets
         ]
+
+        # Pass through potential target merges if the first-order compilation
+        # providers resulted in an empty set of potential target merges.
+        if not potential_target_merges:
+            potential_target_merges = [
+                struct(
+                    src = mergeable_target.src,
+                    dest = id,
+                )
+                for info in deps_infos
+                for mergeable_target in info.direct_potential_target_merges
+            ]
```

Comment:
Does this only go one layer down? Ideally this would go as many layers as needed.

With the `dependencies` handling here: https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/2667f91abe73112f3552543922edd0dec6e6a46b/xcodeproj/internal/target_properties.bzl#L41-L51

we handle this at the level of the target being excluded, not at the top-level (like this)

Reply:
Yup, only one layer. In this case, it's sorta handled by the target _and_ the top-level. I didn't want to join up the second-order potential merge targets unless the first-order resulted in none being present vs. conjoining multiple first-order and potentially second-order merge targets at the same time. Possible that's not a valid concern though.

But let me look at the linked code snippet to see if I can integrate it a different way. 👍